docker/app/src/Controller/GiftcardController.php

```diff
@@ -60,30 +62,57 @@ public function giftget($id): JsonResponse
     /** 
      * @Route("/giftcard/redeem/{id}", methods={"POST"})
      */
-    public function giftredeem($id): JsonResponse
+    public function giftRedeem($id, Request $request): JsonResponse
     {
-        #body money
-        #validacija
+        #u body ubaci money koliko treba oduzet
+        #validacija rjesi da validira dobro sve i da provjeri jeli neko unia -50 npr ili +50
         $firestore = new FirestoreClient([
-            'projectId' => self::projectId,
+            'projectId' => self::PROJECTID,
         ]);
 
+        $newNum = $request->toArray();
+
         $doc = $firestore->collection($_ENV['COLLECTION'])->document($id);
         $snapshot = $doc->snapshot();
 
+        $oldNum = $snapshot->data();
```

Comment:
$oldNum, $newNum naming convection is not a good practice. When you read the code, it should be clear what variables contain.